code/modules/research/experimentor/loot/device_exploot.dm

```diff
@@ -0,0 +1,236 @@
+	..()
+	origin_tech += "biotech=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	origin_tech += "bluespace=[1+raritylevel+rand(raritylevel,raritylevel+1)]"
+	keywords = list("animals", "replication")
+	extra_data = pick(/mob/living/simple_animal/pet/corgi, /mob/living/simple_animal/pet/cat, /mob/living/simple_animal/pet/fox, /mob/living/simple_animal/mouse, /mob/living/simple_animal/pet/pug, /mob/living/simple_animal/lizard, /mob/living/simple_animal/diona, /mob/living/simple_animal/butterfly, /mob/living/carbon/human/monkey)
+
+/obj/item/discovered_tech/floofcannon/itemproc(mob/user)
+	playsound(src.loc, "sparks", rand(25,50), 1)
+	var/mob/living/C = new extra_data(get_turf(user))
+	C.throw_at(pick(oview(10,user)),10,rand(3,8))
+	var/datum/effect_system/smoke_spread/smoke = new
+	smoke.set_up(1,0, user.loc, 0)
+	smoke.start()
+	warn_admins(user, "Floof Cannon", 0)
+
+// Explosion
+/obj/item/discovered_tech/explosion/initialize()
+	..()
+	origin_tech += "materials=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	origin_tech += "combat=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	origin_tech += "plasmatech=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	if(raritylevel>=RARITY_RARE)
+		origin_tech += "toxins=[1+raritylevel+rand(raritylevel,raritylevel+1)]"
+	keywords = list("destruction", "sound")
+
+/obj/item/discovered_tech/explosion/itemproc(mob/user)
+	to_chat(user, "<span class='danger'>[src] begins to heat up!</span>")
+	spawn(rand(max(stability,35),100))
+		if(src.loc == user)
+			visible_message("<span class='notice'>The [src]'s top opens, releasing a powerful blast!</span>")
+			explosion(src.loc, -1, rand(potency/50,potency/30), rand(potency/30,potency/20), rand(potency/20,potency/15), flame_range = potency/25)
+			warn_admins(user, "Explosion")
+			qdel(src)
+
+// Cleaner
+/obj/item/discovered_tech/cleaner/initialize()
+	..()
+	origin_tech += "engineering=[1+raritylevel+rand(raritylevel,raritylevel+1)]"
+	keywords = list("destruction")
+
+/obj/item/discovered_tech/cleaner/itemproc(mob/user)
+	playsound(src.loc, "sparks", rand(25,50), 1)
+	var/obj/item/grenade/chem_grenade/cleaner/CL = new/obj/item/grenade/chem_grenade/cleaner(get_turf(user))
+	CL.prime()
+	warn_admins(user, "Cleaning Foam", 0)
+
+// Flashbang
+/obj/item/discovered_tech/flashbang/initialize()
+	..()
+	origin_tech += "plasmatech=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	keywords = list("light", "sound")
+
+/obj/item/discovered_tech/flashbang/itemproc(mob/user)
+	playsound(src.loc, "sparks", rand(25,50), 1)
+	var/obj/item/grenade/flashbang/CB = new/obj/item/grenade/flashbang(get_turf(user))
+	CB.range = rand(stability/25, max(potency/10, 4))
+	CB.prime()
+	warn_admins(user, "Flashbang")
+
+// Rapid Duplicator
+/obj/item/discovered_tech/rapidDuplicator/initialize()
+	..()
+	origin_tech += "magnets=[1+raritylevel+rand(raritylevel,raritylevel+1)];"
+	origin_tech += "engineering=[1+raritylevel+rand(raritylevel,raritylevel+1)]"
```

Comment:
missing ; at the end of the string. Best to make this into a proc and use it for all techs since you use it so often